src/cascadia/QueryExtension/ILLMProvider.idl

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILLMProvider
```

Comment:
Should any information that the provider wants to share (e.g., about terms, etc.) be part of the interface?

Reply:
Unfortunately not - the terms etc are only relevant for the settings UI and we will not be exposing the LMProvider interface to the settings UI